### PR TITLE
docs: Add missing docs for module(keep)

### LINKED ISSUE #754: Missing docs for `#[rune::function(keep)]`
I have been annoyed for a while that I can't call the Rune extension methods from Rust (when I need them from both locations). I was considering filing a feature request on this, but while looking through rune's own HashMap there is a `keep` attribute that seems to solve this nicely already.

But it is not documented in https://rune-rs.github.io/api/rune/attr.function.html

So: Is the docs missing (in which case I will make a PR) or is it an unstable private API (in which case I propose to stabilise it and document it)?

### PR DESCRIPTION
**Pull request by VorpalBlade**
Fixes #754

I have not made any fix for 0.13. Don't know if we want to backport it (should be very simple if so)

### CODE REVIEW

File: crates/rune/src/lib.rs [hunk 1]
```diff
@@ -577,6 +577,34 @@ pub(crate) use rune_macros::__internal_impl_any;
 /// }
 /// ```
 ///
+/// # Using `keep` to keep the name
+///
+/// By default, the name of the function is mangled and the metadata is given
+/// the original name. This means you can't easily call the function from both
+/// Rune and Rust. This behaviour can be changed by using the `keep` option, in
+/// which case you must  refer to the meta object by a managled name
```
Comment (udoprog) [inline crates/rune/src/lib.rs:585]:
```suggestion
/// which case you must refer to the meta object by a mangled name
```

File: crates/rune/src/lib.rs [hunk 2]
```diff
@@ -577,6 +577,34 @@ pub(crate) use rune_macros::__internal_impl_any;
...
+/// (specifically the function name with `__meta` appended).:
```
Comment (udoprog) [inline crates/rune/src/lib.rs:586]:
```suggestion
/// (specifically the function name with `__meta` appended):
```

File: crates/rune/src/lib.rs [hunk 3]
```diff
@@ -577,6 +577,34 @@ pub(crate) use rune_macros::__internal_impl_any;
 /// }
 /// ```
 ///
+/// # Using `keep` to keep the name
+///
+/// By default, the name of the function is mangled and the metadata is given
+/// the original name. This means you can't easily call the function from both
+/// Rune and Rust. This behaviour can be changed by using the `keep` option, in
```
Comment (udoprog) [inline crates/rune/src/lib.rs:584]:
```suggestion
/// Rune and Rust. This behaviour can be changed by using the `keep` attribute, in
```

**Review by udoprog — CHANGES_REQUESTED** [review]

**Comment by VorpalBlade** [comment]
Merged your changes (on the phone at the moment, and since you apparently squash merge anyway it doesn't matter that I normally try to make clean mergable histories)

**Review by udoprog — APPROVED** [review]

**Comment by udoprog** [comment]
Thanks!

**Pull request merged**